### PR TITLE
[Spark] Improve internal error message for clustering column validation

### PR DESCRIPTION
**Pull request by chirag-s-db**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
Improves error message on internal clustering column API. 

## How was this patch tested?
N/A.

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumn.scala [hunk 1]
```diff
@@ -40,8 +39,12 @@ object ClusteringColumn {
     val physicalNameParts = logicalNameParts.foldLeft[(DataType, Seq[String])]((schema, Nil)) {
       (partial, namePart) =>
         val (currStructType, currPhysicalNameSeq) = partial
-        val field =
-          currStructType.asInstanceOf[StructType].find(field => resolver(field.name, namePart)).get
+        val field = currStructType.asInstanceOf[StructType].find(
+          field => resolver(field.name, namePart)) match {
+          case Some(f) => f
+          case None =>
+            throw DeltaErrors.columnNotInSchemaException(logicalName, schema)
```
Comment (dabao521) [inline spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumn.scala:46]: Can we have a test to validate this exception?

Comment (chirag-s-db) [reply spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumn.scala:47]: Added

**Review by zedtang — APPROVED** [review]
LGTM

File: spark/src/test/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumnSuite.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  test("ClusteringColumn: throws correct error when column not found") {
+    withTable("tbl") {
+      sql("CREATE TABLE tbl (a INT) USING DELTA")
+      val schema = spark.table("tbl").schema
+      val e = intercept[AnalysisException] {
+        ClusteringColumn(schema, "b")
+      }
+      assert(e.getMessage.contains("Couldn't find column b"))
```
Comment (dabao521) [inline spark/src/test/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteringColumnSuite.scala:66]: can we use [checkError](https://github.com/delta-io/delta/blob/1bccf8d906c4ab2109ce071932d6744e0c4b79df/spark/src/test/scala/org/apache/spark/sql/delta/DeltaDropColumnSuite.scala#L452) helper here?

**Review by dabao521 — APPROVED** [review]

**Pull request merged**